### PR TITLE
LPS-170656 Assign the 'ERC' value before Validation step | LPS-179226 Assign 'Uuid' value in 'ERC' field, when this field receive blank string

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issues:
https://issues.liferay.com/browse/LPS-170656;
https://issues.liferay.com/browse/LPS-179226.

### CODE REVIEW

**Comment by p-albuquerque** [comment]
ci:test:relevant

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -3028,26 +3029,31 @@ else if (sqlType == Types.VARCHAR) {
...
 		throws PortalException {
 
 		for (Map.Entry<String, Serializable> entry : values.entrySet()) {
 			if (StringUtil.equals(entry.getKey(), "externalReferenceCode")) {
 				String externalReferenceCode = String.valueOf(entry.getValue());
 
 				if (Validator.isNull(externalReferenceCode)) {
-					externalReferenceCode = String.valueOf(
-						objectEntry.getObjectEntryId());
+					break;
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:3041]: @p-albuquerque aren't we changing the expected behavior by removing this code?

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java [hunk 2]
```diff
@@ -2564,12 +2672,6 @@ private void _testUpdateObjectEntryExternalReferenceCode()
 				"listTypeEntryKeyRequired", "listTypeEntryKey1"
 			).build());
 
-		Assert.assertNotNull(
-			ReflectionTestUtil.getFieldValue(objectEntry1, "_values"));
-		Assert.assertEquals(
-			String.valueOf(objectEntry1.getUuid()),
-			objectEntry1.getExternalReferenceCode());
-
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java:2572]: @p-albuquerque why did you remove this code?

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java [hunk 3]
```diff
@@ -2614,33 +2716,7 @@ private void _testUpdateObjectEntryExternalReferenceCode()
...
-		}
-		catch (ObjectEntryValuesException.MustNotBeDuplicate
-					objectEntryValuesException) {
-
-			Assert.assertEquals(
-				"Duplicate value " +
-					String.valueOf(objectEntry1.getObjectEntryId()),
-				objectEntryValuesException.getMessage());
-		}
+			objectEntry2.getUuid(), objectEntry2.getExternalReferenceCode());
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java:2719]: @p-albuquerque why did you remove this test?

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java [hunk 4]
```diff
@@ -627,6 +628,8 @@ public void testAddObjectEntry() throws Exception {
 					"\"listTypeEntryKeyRequired\"",
 				objectEntryValuesException.getMessage());
 		}
+
+		_testAddObjectEntryExternalReferenceCode();
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java:632]: @p-albuquerque do we really need a new private method? It is only used once

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java [hunk 5]
```diff
@@ -830,6 +833,42 @@ public void testAddObjectEntryWithObjectValidationRule() throws Exception {
...
+					ObjectValidationRuleEngineException.InvalidScript);
+
+			ObjectValidationRuleEngineException.InvalidScript invalidScript =
+				(ObjectValidationRuleEngineException.InvalidScript)
+					modelListenerException.getCause();
+
+			Assert.assertEquals(
+				"there-was-an-error-validating-your-data",
+				invalidScript.getMessageKey());
+		}
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java:871]: @p-albuquerque can we simplify to:

```
catch (ModelListenerException modelListenerException) {
	String message = modelListenerException.getMessage();

	Assert.assertTrue(message.contains("Error"));
}
```
and follow the same pattern in the other tests?

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java [hunk 6]
```diff
@@ -2484,6 +2523,75 @@ private Map<String, Serializable> _getValuesFromDatabase(
 		return values;
 	}
 
+	private void _testAddObjectEntryExternalReferenceCode() throws Exception {
+		ObjectEntry objectEntry1 = _addObjectEntry(
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java:2527]: @p-albuquerque please rename it to `objectEntry` as we don't have another `objectEntry` here

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java (hunk 1 above)
Comment (p-albuquerque) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:3041]: The behavior that it code represent, I believe isn't right, there is an ambiguous behavior around that:
The **ERC** is assigned with **Uuid** value, when we add a _ObjectEntry_ via UI or API(in that case, if we don't explicit a value for ERC).
But, if we are receiving something like this following **valuesMap** of an _ObjectEntry_, the default value assigned in ERC is **objectEntryId** instead **Uuid** :
ERCIssue

Comment (p-albuquerque) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:3041]: besides that, if this scenarios occour while an update, the current ERC will be replaced to **objectEntryId**. With that changes, while update, if some map like this one above will received. The current ERC will be untouched.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java (hunk 2 above)
Comment (p-albuquerque) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java:2572]: I'm doing this same asserts in another place in `testAddObjectEntry`

**Comment by p-albuquerque** [comment]
ci:test:sf

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Review by carolmariaabb — APPROVED** [review]
LGTM!

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:reevaluate:1407401_1068

**Comment by p-albuquerque** [comment]
ci:forward
